superset/common/query_context_processor.py

```diff
@@ -488,21 +511,17 @@ def processing_time_offsets(  # pylint: disable=too-many-locals,too-many-stateme
                 offset_metrics_df = offset_metrics_df.rename(columns=metrics_mapping)
 
                 # 3. set time offset for index
-                index = (get_base_axis_labels(query_object.columns) or [DTTM_ALIAS])[0]
-                if not dataframe_utils.is_datetime_series(offset_metrics_df.get(index)):
-                    raise QueryObjectValidationError(
-                        _(
-                            "A time column must be specified "
-                            "when using a Time Comparison."
-                        )
+                if dataframe_utils.is_datetime_series(offset_metrics_df.get(index)):
+                    # modifies temporal column using offset
+                    offset_metrics_df[index] = offset_metrics_df[index] - DateOffset(
```

Comment:
Hi @Antonio-RiveroMartnez. This logic to calculate the offsets is not correct and I'm fixing it in https://github.com/apache/superset/pull/28432. I think it would be prudent to wait for that fix and rebase your PR to see if it will affect this new feature in any way.

Reply:
Hey @michael-s-molina , thanks for the heads up! Ok, will do, it's very likely that your PR gets merged before this one so it should be ok for rebasing.